src/Controls/src/Core/Platform/Windows/CollectionView/GroupedItemTemplateCollection.cs

```diff
@@ -35,10 +38,17 @@ internal class GroupedItemTemplateCollection : ObservableCollection<GroupTemplat
 			if (_itemsSource is IList groupList && _itemsSource is INotifyCollectionChanged incc)
 			{
 				_groupList = groupList;
-				incc.CollectionChanged += GroupsChanged;
+
+				_collectionChanged = GroupsChanged;
+				_proxy.Subscribe(incc, _collectionChanged);
 			}
 		}
 
+		public void CleanUp()
```

Comment:
If `GroupedItemTemplateCollection` is `internal`, should we just make it `IDisposable` and name this method `Dispose()`?